Remove static_cast in WebCore/platform/graphics

annevk opened the pull request:
#### 0870888ed57f9f62f71a67ccfa071313b29ebcca
<pre>
Remove static_cast in WebCore/platform/graphics
<a href="https://bugs.webkit.org/show_bug.cgi?id=304700">https://bugs.webkit.org/show_bug.cgi?id=304700</a>

Reviewed by Chris Dumez.

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a>

Canonical link: <a href="https://commits.webkit.org/304976@main">https://commits.webkit.org/304976@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/578a67afaf4fc2ffae1bdf4d2e195917f182e417

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/137087 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9447 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48374 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144830 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/90060 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3c4099dd-9c5e-4f92-9fa3-397ec3b18762) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/138959 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/10150 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/9574 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104831 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/76472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a0ffe704-c4ba-458a-8e40-c75471f46b47) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/140032 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7460 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122843 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85666 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e6e6d654-6135-4054-9e80-4cd81d858d10) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/7097 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/4803 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5419 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116445 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/41011 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/147586 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/9130 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41577 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/113185 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/9148 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/7672 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113515 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/7021 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/119107 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/63466 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/21120 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/9178 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/37161 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8902 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/72744 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/9119 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8971 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
```diff
@@ -3946,7 +3946,7 @@ bool GraphicsLayerCA::createFilterAnimationsFromKeyframes(const GraphicsLayerKey
         return false;
 
-    const FilterOperations& operations = static_cast<const GraphicsLayerFilterAnimationValue&>(valueList.at(listIndex)).value();
+    const auto& operations = downcast<const GraphicsLayerFilterAnimationValue>(valueList.at(listIndex)).value();
```
I think we can drop both `const`

cdumez commented on Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
```diff
@@ -3842,7 +3842,7 @@ bool GraphicsLayerCA::appendToUncommittedAnimations(const GraphicsLayerKeyframeV
 static const TransformOperations& transformationAnimationValueAt(const GraphicsLayerKeyframeValueList& valueList, unsigned i)
 {
-    return static_cast<const GraphicsLayerTransformAnimationValue&>(valueList.at(i)).value();
+    return downcast<const GraphicsLayerTransformAnimationValue>(valueList.at(i)).value();
```
I think we can drop the `const`

cdumez commented on Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
```diff
@@ -4057,8 +4057,8 @@ bool GraphicsLayerCA::setAnimationEndpoints(const GraphicsLayerKeyframeValueList
     case AnimatedProperty::Opacity: {
-        basicAnim->setFromValue(static_cast<const GraphicsLayerFloatAnimationValue&>(valueList.at(fromIndex)).value());
-        basicAnim->setToValue(static_cast<const GraphicsLayerFloatAnimationValue&>(valueList.at(toIndex)).value());
+        basicAnim->setFromValue(downcast<const GraphicsLayerFloatAnimationValue>(valueList.at(fromIndex)).value());
```
I think we can drop the `const`.

cdumez commented on Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
```diff
@@ -4057,8 +4057,8 @@ bool GraphicsLayerCA::setAnimationEndpoints(const GraphicsLayerKeyframeValueList
-        basicAnim->setFromValue(static_cast<const GraphicsLayerFloatAnimationValue&>(valueList.at(fromIndex)).value());
-        basicAnim->setToValue(static_cast<const GraphicsLayerFloatAnimationValue&>(valueList.at(toIndex)).value());
+        basicAnim->setFromValue(downcast<const GraphicsLayerFloatAnimationValue>(valueList.at(fromIndex)).value());
+        basicAnim->setToValue(downcast<const GraphicsLayerFloatAnimationValue>(valueList.at(toIndex)).value());
```
I think we can drop the `const`.

cdumez commented on Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
```diff
@@ -4084,7 +4084,7 @@ bool GraphicsLayerCA::setAnimationKeyframes(const GraphicsLayerKeyframeValueList
         switch (valueList.property()) {
         case AnimatedProperty::Opacity: {
-            const auto& floatValue = static_cast<const GraphicsLayerFloatAnimationValue&>(curValue);
+            const auto& floatValue = downcast<const GraphicsLayerFloatAnimationValue>(curValue);
```
I think we can drop both `const`.

cdumez commented on Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
```diff
@@ -4174,7 +4174,7 @@ bool GraphicsLayerCA::setTransformAnimationKeyframes(const GraphicsLayerKeyframe
     for (unsigned i = 0; i < valueList.size(); ++i) {
         unsigned index = forwards ? i : (valueList.size() - i - 1);
-        const auto& curValue = static_cast<const GraphicsLayerTransformAnimationValue&>(valueList.at(index));
+        const auto& curValue = downcast<const GraphicsLayerTransformAnimationValue>(valueList.at(index));
```
I think we can drop both `const`.

cdumez commented on Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
```diff
@@ -4234,8 +4234,8 @@ bool GraphicsLayerCA::setFilterAnimationEndpoints(const GraphicsLayerKeyframeVal
 
-    const auto& fromValue = static_cast<const GraphicsLayerFilterAnimationValue&>(valueList.at(fromIndex));
-    const auto& toValue = static_cast<const GraphicsLayerFilterAnimationValue&>(valueList.at(toIndex));
+    const auto& fromValue = downcast<const GraphicsLayerFilterAnimationValue>(valueList.at(fromIndex));
```
ditto

cdumez commented on Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
```diff
@@ -4234,8 +4234,8 @@ bool GraphicsLayerCA::setFilterAnimationEndpoints(const GraphicsLayerKeyframeVal
-    const auto& fromValue = static_cast<const GraphicsLayerFilterAnimationValue&>(valueList.at(fromIndex));
-    const auto& toValue = static_cast<const GraphicsLayerFilterAnimationValue&>(valueList.at(toIndex));
+    const auto& fromValue = downcast<const GraphicsLayerFilterAnimationValue>(valueList.at(fromIndex));
+    const auto& toValue = downcast<const GraphicsLayerFilterAnimationValue>(valueList.at(toIndex));
```
ditto

cdumez commented on Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
```diff
@@ -4272,7 +4272,7 @@ bool GraphicsLayerCA::setFilterAnimationKeyframes(const GraphicsLayerKeyframeVal
     for (unsigned i = 0; i < valueList.size(); ++i) {
         unsigned index = forwards ? i : (valueList.size() - i - 1);
-        const auto& curValue = static_cast<const GraphicsLayerFilterAnimationValue&>(valueList.at(index));
+        const auto& curValue = downcast<const GraphicsLayerFilterAnimationValue>(valueList.at(index));
```
ditto

annevk committed:
Remove static_cast in WebCore/platform/graphics
https://bugs.webkit.org/show_bug.cgi?id=304700

Reviewed by Chris Dumez.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

Canonical link: https://commits.webkit.org/304976@main
